### PR TITLE
Add debug and local provider

### PR DESCRIPTION
**Pull request by CasperWA**
Updates:
- The general purpose notebook `OPTiMaDe general.ipynb` has been updated with an output section.
- Split up `helper_widgets.py` into separate files under the module `subwigets`.
- Top-level QueryWidget now supports AiiDA `StructureData` as well as `ase.Atoms`.
- Filter inputs are now "frozen" until a database is chosen.
- Prior text output is now _only_ used for error messages (and removed/reset upon new queries).
- Formula is now `hill_compact`.
- Using providers.optimade.org instead of optimade.org/providers to retrieve list of Materials-Consortia OPTiMaDe providers.

New:
- Added a "top-level ResultsWidget.
- `ParserError`, which may be used to provide helpful information about wrong user inputs in the filter inputs.
- `embedded` parameter for QueryWidget, to understand whether or not the widget is used as part of another widget or not.
- Results widget `StructureDataSites` shows a list of sites of the given structure.
  This is based on the AiiDA `Dict` viewer in `aiidalab-widgets-base`.
- Hard-coded supported OPTiMaDe version.
- Attempt to retrieve implementations where they combine an index meta-database and implementation in the same base URL.
- Use versioned base URLs to retrieve implementations.
- Attempt to better handle non-JSON responses when performing the query.
- Remove dependency on `aiidalab` and instead depend on `aiidalab-widgets-base` for the structure viewer.
- If in debug-mode, a local server is used (must be running on `http://localhost:5000/optimade/vMAJOR`, where `MAJOR` represent the major version number of the supported OPTiMaDe version, at this point `v0`, since the supported version is `0.10.1`.

### CODE REVIEW

File: aiidalab_optimade/subwidgets/output_summary.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+class StructureDataSites(ipw.HTML):
+    """StructureData Sites Output
+    Reimplements the viewer for AiiDA Dicts
+    """
+
+    structure = traitlets.Instance(StructureData, allow_none=True)
+
+    def __init__(self, structure: StructureData = None, **kwargs):
+        # For more information on how to controle the table appearance please visit:
```
Comment (CasperWA) [inline aiidalab_optimade/subwidgets/output_summary.py]:
```suggestion
        # For more information on how to control the table appearance please visit:
```

File: aiidalab_optimade/subwidgets/output_summary.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    def _on_change_structure(self, change: dict):
+        """When traitlet 'sites' is updated"""
+        if change["new"] is None:
+            self.value = ""
+        else:
+            self.value = self._style
+            dataf = pd.DataFrame(
+                self._format_sites(), columns=["Elements", "Occypancy", "Position"]
+            )
+            # self.value += dataf.to_string()
```
Comment (CasperWA) [inline aiidalab_optimade/subwidgets/output_summary.py]:
```suggestion
```

**Comment by giovannipizzi** [comment]
Wait... do you seriously self-review your own PRs? :-D

**Comment by CasperWA** [comment]
> Wait... do you seriously self-review your own PRs? :-D

... sometimes... :smirk:

**Pull request merged**